core/domain/exp_domain.py

```diff
@@ -1013,7 +1013,18 @@ def has_state_name(self, state_name):
             bool. Returns true if the exploration has the given state name.
         """
         state_names = self.states.keys()
-        return bool(state_name in state_names)
+        return state_name in state_names
+
+    def get_interaction_id_by_state_name(self, state_name):
+        """Returns the interaction id of the state.
+
+        Args:
+            state_name: str. The name of the state.
+
+        Returns:
+            str. The ID of the interaction.
+        """
+        return self.states[state_name].interaction.id
```

Comment:
This could be None right? Should the callers take care of this? Also should the docstring indicate this?

Reply:
Yeah this can be None, I have modified the docstring accordingly.

Reply:
Thanks for raising this @nithusha21.

@anubhavsinha98, do make sure we are handling the NOne case correctly in the controller/service layer.

Reply:
Yeah @DubeySandeep the None case is being handled in the validation part of the domain object.